2.preprocess_features/scripts/0.convert_cytotable.py

```diff
@@ -0,0 +1,161 @@
+
+# merge single cells and output as parquet file
+convert(
+    source_path=str(sqlite_dir),
+    dest_path=str(parquet_path),
+    dest_datatype=dest_datatype,
+    metadata=["image"],
+    compartments=["mito", "lamellar", "tubular"],
+    identifying_columns=["ImageNumber"],
+    join=False,
+    parsl_config=Config(
+        executors=[HighThroughputExecutor()],
+    ),
+    chunk_size=10000,
+)
+
+print("Conversion finished for EM data")
+
+
+# ## Move parquet files out of folders with the same name for easier access in downstream analysis
+
+# In[4]:
+
+
+# Edit the structure of the output to remove the extra nested folders in each plate folder
+for parent_dir in output_dir.iterdir():
+    if parent_dir.is_dir():
+        for subdir in parent_dir.iterdir():
+            if subdir.is_dir():
+                for file_path in subdir.glob("*.parquet"):
+                    new_path = parent_dir / file_path.name
+                    file_path.rename(new_path)
+                # Remove the subfolder with the same name as the parquet file
+                subdir.rmdir()
+
+print("Parquet files moved down in the nest and folders removed.")
+
+
+# ## Remove unwanted image + metadata columns and merge relevant metadata into each mask/object file from the image file
+
+# In[5]:
+
+
+# path to unwanted image cols text file
+unwanted_list_path = pathlib.Path("./unwanted_image_cols.txt")
+# Load the list of columns to remove from the text file
+with open(unwanted_list_path, "r") as file:
+    columns_to_remove = [line.strip() for line in file]
+
+
+# In[6]:
+
+
+# Read in each file as data frame
+image_df = pd.read_parquet(pathlib.Path(f"{parquet_path}/per_image.parquet"))
+mito_df = pd.read_parquet(pathlib.Path(f"{parquet_path}/per_mito.parquet"))
+lamellar_df = pd.read_parquet(pathlib.Path(f"{parquet_path}/per_lamellar.parquet"))
+tubular_df = pd.read_parquet(pathlib.Path(f"{parquet_path}/per_tubular.parquet"))
+
+print(
+    "Starting to edit image and object data frames for EM data",
+)
+
+# Drop the specified columns from image_df (ignore error if a column isn't there)
+image_df = image_df.drop(columns=columns_to_remove)
+
+# Identify metadata columns for mask data frame
+metadata_cols = [
+    col
+    for col in image_df.columns
+    if col.startswith("Image_Metadata") or col == "Metadata_ImageNumber"
+]
+
+# Change prefix and remove "Image_" portion from metadata columns
+new_metadata_cols = [col.replace("Image_", "") for col in metadata_cols]
+image_df = image_df.rename(columns=dict(zip(metadata_cols, new_metadata_cols)))
+
+# Merge metadata columns from image_df to each object data frame based on Metadata_ImageNumber
+mito_df = pd.merge(
+    mito_df, image_df[new_metadata_cols], on="Metadata_ImageNumber", how="inner"
+)
+lamellar_df = pd.merge(
+    lamellar_df, image_df[new_metadata_cols], on="Metadata_ImageNumber", how="inner"
+)
+tubular_df = pd.merge(
+    tubular_df, image_df[new_metadata_cols], on="Metadata_ImageNumber", how="inner"
+)
```

Comment:
I believe pycytominer's `annotate` function can accomplish this task. According to the [documentation](https://pycytominer.readthedocs.io/en/latest/pycytominer.html#pycytominer.annotate.annotate), you can use it as follows:

```python
from pycytominer import annotate

annotate(profiles=mito_df, platemap=image_df[new_metadata_cols], join_on=["Metadata_ImageNumber", "Metadata_ImageNumber"])
```

Looking at the `annotate()` source [code](https://github.com/cytomining/pycytominer/blob/c782728434ac10ce65f72db07402a0664d488f72/pycytominer/annotate.py#L85), it performs an `inner` join, similar to your code!

Reply:
if it doesn't work feel free to ignore this suggestion c:

Reply:
WOAH! I didn't know I could do that! Just updated this and that saved me a bunch of lines of code where I was moving the metadata around. Thanks Erik!